stdlib/REPL/src/docview.jl

```diff
@@ -624,22 +624,61 @@ bestmatch(needle, haystack) =
     longer(matchinds(needle, haystack, acronym = true),
            matchinds(needle, haystack))
 
-avgdistance(xs) =
-    isempty(xs) ? 0 :
-    (xs[end] - xs[1] - length(xs)+1)/length(xs)
+# Optimal string distance: Counts the minimum number of insertions, deletions,
+# transpositions or substitutions to go from one string to the other.
+function string_distance(a::AbstractString, lena::Integer, b::AbstractString, lenb::Integer)
```

Comment:
Is this the same as the [Levenshtein distance](https://en.wikipedia.org/wiki/Levenshtein_distance)?  Should this be exported somewhere?

Reply:
Not quite, string distance also counts a transposition as a single edit

Reply:
IIRC some investigation estimated the number of typos that are transpositions at ~13%, making them well worth accounting for.